sickbeard/properFinder.py

```diff
@@ -214,10 +215,22 @@ def _getProperList(self):  # pylint: disable=too-many-locals, too-many-branches,
             # only keep the proper if we have already retrieved the same quality ep (don't get better/worse ones)
             oldStatus, oldQuality = Quality.splitCompositeStatus(int(sql_results[0]['status']))
             if oldStatus not in (DOWNLOADED, SNATCHED) or oldQuality != bestResult.quality:
-                logger.log(u"Ignoring proper already snatched or downloaded: {name}".format
+                logger.log(u"Ignoring proper because quality is different or episode is already archived: {name}".format
                            (name=bestResult.name))
                 continue
 
+            # check if we already snatched that PROPER
+            main_db_con = db.DBConnection()
+            sql_results = main_db_con.select("SELECT resource FROM history WHERE (action LIKE '%2' OR action LIKE '%9') \
+                                                AND showid = ? AND season = ? AND episode = ?",
+                                                [bestResult.indexerid, bestResult.season, bestResult.episode])
+            if sql_results:
+                new_proper = remove_strings(remove_non_release_groups(bestResult.name), ['.mkv','.avi','.mp4'])
+                if any(result for result in sql_results if result['resource'] 
+                        and remove_strings(remove_non_release_groups(str(result['resource'])), ['.mkv','.avi','.mp4']) == new_proper):
```

Comment:
can you explain why wee need to remove all these strings, just to compare if the release/resource was already in the history table? Why not just a simple compare resource, without all stripping?